src/java/org/apache/cassandra/db/compaction/UnifiedCompactionStrategy.java

```diff
@@ -98,7 +98,7 @@ public String toString()
 
     public UnifiedCompactionStrategy(CompactionStrategyFactory factory, BackgroundCompactions backgroundCompactions, Map<String, String> options)
     {
-        this(factory, backgroundCompactions, options, Controller.fromOptions(factory.getRealm(), options));
+        this(factory, backgroundCompactions, options, Controller.fromOptions(factory.getRealm(), options, factory.getRealm().metadata().hasVectorType()));
```

Comment:
As `fromOptions` has access to the realm, why not let it get the `hasVectorType` on its own? (And mock it with `when(cfs.hasVectorType()).thenAnswer(...);` in `ControllerTest`.)